jenkins/image_building/build-ci-image.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+set -eux
+
+export IMAGE_OS="${IMAGE_OS}"
+
+current_dir="$(dirname "$(readlink -f "${0}")")"
+
+# Disable needrestart interactive mode
+sudo sed -i "s/^#\$nrconf{restart} = 'i';/\$nrconf{restart} = 'a';/" /etc/needrestart/needrestart.conf   > /dev/null
+
+sudo apt-get update
+
+# Install packages
+sudo apt-get install python3-pip qemu qemu-kvm -y
+sudo pip3 install diskimage-builder python-openstackclient
+
+# shellcheck disable=SC1091
+. "$current_dir"/dib-and-image-vars.sh
```

Comment:
I don't think this is used anywhere else. Why do we need a separate file for it?

Reply:
for better readability?

Reply:
IMO it makes the code less readable. For example if one sees ${FINAL_CI_IMG_NAME} at the end of the build file, and wonder where it was defined, they would have to use grep to find it, since there's no clue in the code suggesting where that was defined.

I know we did this kind of sourcing very frequently in our CI and dev-env, but it's really a practice we should think twice before applying. In this case I see no value in it, so I'd vote to move all of the export to the outer script and remove the sourced file.

Another nit: I'm not sure how many of these variables need to be exported, but if some of them do not, then please consider removing the `export` keyword.

Reply:
`

> IMO it makes the code less readable. For example if one sees ${FINAL_CI_IMG_NAME} at the end of the build file, and wonder where it was defined, they would have to use grep to find it, since there's no clue in the code suggesting where that was defined.
> 
> I know we did this kind of sourcing very frequently in our CI and dev-env, but it's really a practice we should think twice before applying. In this case I see no value in it, so I'd vote to move all of the export to the outer script and remove the sourced file.

Fair. I removed the dib-and-image-vars.sh file.

Reply:
> Can you be more specific which keywords are unnecessarily exported?

Reply:
Nothing. I'm just mentioning it, just in case some of them are not needed. (I have no idea if that's the case).